frontend/src/Series/MonitoringOptions/MonitoringOptionsModalContent.js

```diff
@@ -0,0 +1,140 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { inputTypes } from 'Helpers/Props';
+import Button from 'Components/Link/Button';
+import SpinnerButton from 'Components/Link/SpinnerButton';
+import ModalContent from 'Components/Modal/ModalContent';
+import ModalHeader from 'Components/Modal/ModalHeader';
+import ModalBody from 'Components/Modal/ModalBody';
+import ModalFooter from 'Components/Modal/ModalFooter';
+import Form from 'Components/Form/Form';
+import FormGroup from 'Components/Form/FormGroup';
+import FormLabel from 'Components/Form/FormLabel';
+import FormInputGroup from 'Components/Form/FormInputGroup';
+
+const NO_CHANGE = 'noChange';
+
+class MonitoringOptionsModalContent extends Component {
+
+  //
+  // Lifecycle
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      monitor: NO_CHANGE
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      isSaving,
+      saveError
+    } = prevProps;
+
+    if (prevProps.isSaving && !isSaving && !saveError) {
+      this.setState({
+        monitor: NO_CHANGE
+      });
+    }
+  }
+
+  onInputChange = ({ name, value }) => {
+    this.setState({ [name]: value });
+  }
+
+  //
+  // Listeners
+
+  onSavePress = () => {
+    const {
+      onSavePress
+    } = this.props;
+    const {
+      monitor
+    } = this.state;
+
+    const changes = {};
+
+    if (monitor !== NO_CHANGE) {
+      changes.monitor = monitor;
+      onSavePress(changes);
```

Comment:
`onSavePress({ monitor });` and drop the `changes` constant.

Reply:
This is sorted